call_server/campaign/views.py

```diff
@@ -189,7 +189,9 @@ def upload_recording(campaign_id):
         # save uploaded file to storage
         file_storage = request.files.get('file_storage')
         if file_storage:
-            file_storage.filename = "campaign_{}_{}_{}.mp3".format(campaign.id, message_key, recording.version)
+            extension = "mp3" if file_storage.mimetype == "audio/mp3" else "wav"
```

Comment:
There was a fix for the file extension issue upstream.

https://github.com/OpenSourceActivismTech/call-power/commit/d32c09989e6b548332f1e8ccbce98ac74c538628#diff-eb999b3a59ac05321a5d60b47f513abeR191

Can we cherry-pick it?

Reply:
Any idea why commit adds parses the extension on the front end, puts it in a hidden field, and sends it to the backend?

It seems more straightforward and less trusting of user input to just grab the extension from the filename on the backend. That might be better/lead to fewer surprising outcomes than basing it on the mime type. 

Are we trying to stay more or less in sync with that fork? If so I'll cherry-pick it over. If not and you agree with the point above I'll update this to grab the extension on the backend.

Reply:
We have an issue (long in the backlog) to pull changes from that repo to get some improvements to the way campaign statistics are tallied. Right now we can merge with [no conflicts](https://github.com/EFForg/call-congress/compare/master...OpenSourceActivismTech:master), but I don't mind resolving a few during the upgrade if they're improvements that we can try to push upstream afterwards. We did that last time and they were merged pretty readily.

I agree it'd be easier to split the extension off on the backend. I'm not sure why there's a form field for it. Sometimes the upload is a `Blob` recorded from the microphone, which doesn't have a filename, but it looks like you can provide the name manually in [`formData.append`](https://developer.mozilla.org/en-US/docs/Web/API/FormData/append).

Reply:
I'd be happy to accept a PR for this new functionality, which looks much smarter than my roundabout way of parsing the file on the frontend.

Reply:
Cool, I'd be glad to open one. This PR also does a deeper filetype check using a python wrapper for the Linux `file` command. It's fast but adds a system dependency (which would normally be present on a Linux system. it was in the docker container already, for example). Would you be interested in adding that or just moving the extension check to the backend?

We added the deeper check because we were running into issues with users uploading file that didn't match their headers, but I don't know if that's a problem outside our org.